firebase-dataconnect/src/androidTest/kotlin/com/google/firebase/dataconnect/GrpcMetadataIntegrationTest.kt

```diff
@@ -165,6 +174,64 @@ class GrpcMetadataIntegrationTest : DataConnectIntegrationTestBase() {
     verifyMetadataDoesNotContain(metadatasJob2, firebaseAuthTokenHeader)
   }
 
+  @Test
+  fun executeQueryShouldNotSendAppCheckMetadataWhenAppCheckIsNotEnabled() = runTest {
+    // TODO: Add an integration test where the AppCheck dependency is absent, and ensure that no
+    // appcheck token is sent at all.
+    val grpcServer = inProcessDataConnectGrpcServer.newInstance()
+    val dataConnect = dataConnectFactory.newInstance(grpcServer)
+    val queryRef = dataConnect.query("qrybbeekpkkck", Unit, serializer<Unit>(), serializer<Unit>())
+    val metadatasJob = async { grpcServer.metadatas.first() }
+
+    queryRef.execute()
+
+    verifyMetadataContains(metadatasJob, firebaseAppCheckTokenHeader, PLACEHOLDER_APP_CHECK_TOKEN)
```

Comment:
Should this be `DoesNotContain`?

Reply:
Oh I should rename this (and the next) test. Since the AppCheck SDK is a gradle dependency then it sends a "placeholder" token if AppCheck is not configured because that's what the AppCheck SDK gives to Data Connect. If the AppCheck SDK is _not_ a gradle dependency, then it would indeed omit the token altogether. I've added a "TODO" comment above to do just that. Hope that makes sense.

Reply:
Ahh, that makes perfect sense. Thanks!